too many levels of symbolic links crash.

yes, its an error in my Dockerfile, but I presume you don't want to panic.

Docker version 1.12.2-rc3, build cb0ca64, experimental

my Dockerfile had:

```
RUN ln -s /sbin/getty /sbin/agetty
...
COPY agetty /sbin/agetty
```

but it didn't crash until the `RUN` line after the `COPY`.

```
Step 8 : RUN addgroup -g 1100 rancher &&     addgroup -g 1101 docker &&     addgroup -S sudo &&     adduser -D -u 1100 -g rancher -G docker -G sudo -s /bin/bash rancher &&     adduser -D -u 1101 -g docker -G docker -G sudo -s /bin/bash docker &&     echo ClientAliveInterval 180 >> /etc/ssh/sshd_config &&     echo '## allow password less for rancher user' >> /etc/sudoers &&     echo 'rancher ALL=(ALL) NOPASSWD: ALL' >> /etc/sudoers &&     echo '## allow password less for docker user' >> /etc/sudoers &&     echo 'docker ALL=(ALL) NOPASSWD: ALL' >> /etc/sudoers &&     echo '%sudo ALL=(ALL) ALL%sudo ALL=(ALL) ALL'  >> /etc/sudoers &&     ln -sf /usr/bin/docker.dist /usr/bin/docker &&     ln -sf /usr/bin/docker-containerd.dist /usr/bin/docker-containerd &&     ln -sf /usr/bin/docker-containerd-shim.dist /usr/bin/docker-containerd-shim &&     ln -sf /usr/bin/docker-runc.dist /usr/bin/docker-runc     echo 'ssh-keygen -A' >> /usr/sbin/update-ssh-keys
 ---> Running in 8250d0d1b6f3
panic: standard_init_linux.go:175: exec user process caused "too many levels of symbolic links" [recovered]
    panic: standard_init_linux.go:175: exec user process caused "too many levels of symbolic links"

goroutine 1 [running, locked to thread]:
panic(0x7eacc0, 0xc8201372d0)
    /usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/urfave/cli.HandleAction.func1(0xc8201092e8)
    /go/src/github.com/opencontainers/runc/Godeps/_workspace/src/github.com/urfave/cli/app.go:478 +0x38e
panic(0x7eacc0, 0xc8201372d0)
    /usr/local/go/src/runtime/panic.go:443 +0x4e9
github.com/opencontainers/runc/libcontainer.(*LinuxFactory).StartInitialization.func1(0xc820108bf8, 0xc82001e0d0, 0xc820108d08)
    /go/src/github.com/opencontainers/runc/Godeps/_workspace/src/github.com/opencontainers/runc/libcontainer/factory_linux.go:259 +0x136
github.com/opencontainers/runc/libcontainer.(*LinuxFactory).StartInitialization(0xc820069630, 0x7f589e560728, 0xc8201372d0)
    /go/src/github.com/opencontainers/runc/Godeps/_workspace/src/github.com/opencontainers/runc/libcontainer/factory_linux.go:277 +0x5b1
main.glob.func8(0xc820086a00, 0x0, 0x0)
    /go/src/github.com/opencontainers/runc/main_unix.go:26 +0x68
reflect.Value.call(0x74f5c0, 0x900fe8, 0x13, 0x847308, 0x4, 0xc820109268, 0x1, 0x1, 0x0, 0x0, ...)
    /usr/local/go/src/reflect/value.go:435 +0x120d
reflect.Value.Call(0x74f5c0, 0x900fe8, 0x13, 0xc820109268, 0x1, 0x1, 0x0, 0x0, 0x0)
    /usr/local/go/src/reflect/value.go:303 +0xb1
github.com/urfave/cli.HandleAction(0x74f5c0, 0x900fe8, 0xc820086a00, 0x0, 0x0)
    /go/src/github.com/opencontainers/runc/Godeps/_workspace/src/github.com/urfave/cli/app.go:487 +0x2ee
github.com/urfave/cli.Command.Run(0x84a1c8, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x8e02a0, 0x51, 0x0, ...)
    /go/src/github.com/opencontainers/runc/Godeps/_workspace/src/github.com/urfave/cli/command.go:191 +0xfec
github.com/urfave/cli.(*App).Run(0xc820001b00, 0xc82000a100, 0x2, 0x2, 0x0, 0x0)
    /go/src/github.com/opencontainers/runc/Godeps/_workspace/src/github.com/urfave/cli/app.go:240 +0xaa4
main.main()
    /go/src/github.com/opencontainers/runc/main.go:137 +0xe24
```

--- a/libcontainer/factory_linux.go
+++ b/libcontainer/factory_linux.go
@@ -256,11 +256,13 @@ func (l *LinuxFactory) StartInitialization() (err error) {
 		if _, ok := i.(*linuxStandardInit); ok {
 			//  Synchronisation only necessary for standard init.
 			if werr := utils.WriteJSON(pipe, syncT{procError}); werr != nil {
-				panic(err)
+				fmt.Fprintln(os.Stderr, err)
+				return
 			}
 		}
 		if werr := utils.WriteJSON(pipe, newSystemError(err)); werr != nil {
-			panic(err)
+			fmt.Fprintln(os.Stderr, err)
+			return
 		}
 		// ensure that this pipe is always closed
 		pipe.Close()

